packages/insomnia/src/ui/components/dropdowns/preview-mode-dropdown.tsx

```diff
@@ -41,6 +43,64 @@ export const PreviewModeDropdown: FC<Props> = ({
     copyToClipboard();
   }, [copyToClipboard]);
 
+  const exportAsHAR = async () => {
+    if (!response || !request || !isRequest(request)) {
+      console.warn('Nothing to download');
+      return;
+    }
+
+    const data = await exportHarCurrentRequest(request, response);
+    const har = JSON.stringify(data, null, '\t');
+
+    const { filePath } = await window.dialog.showSaveDialog({
+      title: 'Export As HAR',
+      buttonLabel: 'Save',
+      defaultPath: `${request.name.replace(/ +/g, '_')}-${Date.now()}.har`,
+    });
+
+    if (!filePath) {
+      return;
+    }
+    const to = fs.createWriteStream(filePath);
+    to.on('error', err => {
+      console.warn('Failed to export har', err);
+    });
+    to.end(har);
+  };
+
+  const exportDebugFile = async () => {
+    if (!response || !request) {
+      console.warn('Nothing to download');
+      return;
+    }
+
+    const timeline = models.response.getTimeline(response);
+    const headers = timeline
+      .filter(v => v.name === 'HeaderIn')
+      .map(v => v.value)
+      .join('');
+
+    const { canceled, filePath } = await window.dialog.showSaveDialog({
+      title: 'Save Full Response',
+      buttonLabel: 'Save',
+      defaultPath: `${request.name.replace(/ +/g, '_')}-${Date.now()}.txt`,
+    });
+
+    if (canceled) {
+      return;
+    }
+    const readStream = models.response.getBodyStream(response);
+
+    if (readStream && filePath) {
```

Comment:
thought:

before, this line of code read:
```suggestion
    if (readStream) {
```

Which means that `filePath`, if set to emptyString (`''`) would be passed to `fs.createWriteStream`.  I tried it locally, and it  gave me `<ref *1> [Error: ENOENT: no such file or directory, open ''] {`, which I _think_ means we're not introducing a bug by doing this (my thinking was, perhaps in a situation where the empty string would be taken to mean PWD or something).

Reply:
this was added in order to remove `// @ts-expect-error -- TSCONVERSION`

Reply:
yep, I get that, but in this case it is a logical change so I wanted to make sure the logical change doesn't have unexpected consequences and I wanted to take note that I followed up on it